tests/jax/components/updating/parameter_server_test.py

```diff
@@ -0,0 +1,233 @@
+    test_default_parameter_server.config.checkpoint = True
+    test_default_parameter_server.on_parameter_server_init_start(server)
+
+    assert server.store.last_checkpoint_time == 0
+    assert hasattr(server.store, "system_checkpointer")
+    # Test nothing more for now, since it's weird that a tf checkpointer is being used
+
+
+def test_on_parameter_server_get_parameters_single(
+    test_default_parameter_server: DefaultParameterServer, server: SystemParameterServer
+) -> None:
+    """Test get_parameters when only a single parameter is requested"""
+    server.store._param_names = "param2"
+
+    test_default_parameter_server.on_parameter_server_get_parameters(server)
+
+    assert server.store.get_parameters == "param2_value"
+
+
+def test_on_parameter_server_get_parameters_list(
+    test_default_parameter_server: DefaultParameterServer, server: SystemParameterServer
+) -> None:
+    """Test get_parameters when a list of parameters are requested"""
+    server.store._param_names = ["param1", "param3"]
+
+    test_default_parameter_server.on_parameter_server_get_parameters(server)
+
+    assert server.store.get_parameters["param1"] == "param1_value"
+    assert server.store.get_parameters["param3"] == "param3_value"
+    assert "param2" not in server.store.get_parameters.keys()
+
+
+def test_on_parameter_server_set_parameters(
+    test_default_parameter_server: DefaultParameterServer, server: SystemParameterServer
+) -> None:
+    """Test setting parameters"""
+    server.store._set_params = {
+        "param1": "param1_new_value",
+        "param3": "param3_new_value",
+    }
+
+    test_default_parameter_server.on_parameter_server_set_parameters(server)
+
+    assert server.store.parameters["param1"] == "param1_new_value"
+    assert server.store.parameters["param2"] == "param2_value"
+    assert server.store.parameters["param3"] == "param3_new_value"
+
+
+def test_on_parameter_server_add_to_parameters(
+    test_default_parameter_server: DefaultParameterServer, server: SystemParameterServer
+) -> None:
+    """Test addition on parameters"""
+    server.store.parameters["param3"] = 4
+    server.store._add_to_params = {
+        "param1": "_param1_add",
+        "param3": 2,
+    }
+
+    test_default_parameter_server.on_parameter_server_add_to_parameters(server)
+
+    assert server.store.parameters["param1"] == "param1_value_param1_add"
+    assert server.store.parameters["param2"] == "param2_value"
+    assert server.store.parameters["param3"] == 6
+
+
+def test_on_parameter_server_run_loop(
+    test_default_parameter_server: DefaultParameterServer, server: SystemParameterServer
+) -> None:
+    """Test checkpointing in run loop"""
+    server.store.last_checkpoint_time = 0
+
+    # Do nothing if no checkpointer
+    test_default_parameter_server.config.checkpoint = False
+    assert (
+        server.store.last_checkpoint_time
+        + test_default_parameter_server.config.checkpoint_minute_interval * 60
+        + 1
```

Comment:
Please just putting a comment here explaining what this represents, a bit confusing to work out